Fix for the issue #333

alifirat opened the pull request:
I suggest the following fix (bases on the comments of @ipfix ) :
- I added a new configuration variable called "cancel_private_links" in order to check if a user wants to cancel the links for the names "xx"
- For a wizard or a friend, nothing is canceled because they should see the private data.
- For a visitor, if the new variable is setting to yes we just remove the link and otherwise, the link is still here. 

Hope this is a correct fix !

fablhx committed:
Do not check if parents has title to compute the authorized age.

alifirat committed:
Add the new variable cancel_private_links in the configuration type

alifirat committed:
Change the displaying of a name in order to remove the link when the options "hide_private_names" and "cancel_private_links" are set to yes

alifirat committed:
Add a new line in order to get the value of "cancel_private_links" in a configuration file

alifirat committed:
Add a new variable case in the pattern for the cancel_private_links var in the function evaluating variables value

alifirat committed:
Add a file in order to fix for the issue 333

alifirat committed:
Add a file in order to test the fix for the issue ###333

alifirat committed:
Merge branch 'cancel_private_links##issue333' of github.com:alifirat/geneweb into cancel_private_links##issue333

swdes commented:
Hello alifirat 
Thank you for the submission. I haven't test your pull request yet but I've a few comments:
- maybe we should add ";" after %xx as everywhere. Also you should remove <a>. In fact, why don't you use the above condition ? for something like that:

```
%if;(cancel_links or xx.is_restricted or (xx = "x x" and cancel_private_links))%xx;
```
- are you sure there's not a prettier way to test xx ? like xx.is_obfuscated ?
- about the test files, could you keep to the current ones (grimaldi.*) and suggest an addition to the .gwf

alifirat commented:
<uI>
<li> I didn't add a ";" after %xx instruction because the ";" is displaying in the HTML page too </li>
<li> It's true that I can join the condition that I added to the previous condition in the function like you do </li>
<li>I tested the xx value like this because I saw similar examples in the file but I also will change that.</li>
<li>Yes, I will remove the test.\* and add my suggestion to the grimaldi.\* files </li>
</ul> 

Thank you for replying and suggestions, I change the code in the day.

alifirat committed:
Add an expression to the condition showing "x x" without links (I add the case when the cancel_private_links is set to yes)

alifirat committed:
Add the New case "is_obfuscated" in the pattern representing a private name

alifirat committed:
Add a new variable to cancel the private links

alifirat commented:
I update the PR with new commits : 

<ul>
<li>Instead of add a new instruction to cancel private links, I updated the condition with a new case like in your suggestions.</li>
<li>To check the value of  xx, I added a new field for a person called "is_obfuscated" checking if the field "hide_private_names" is set to yes (this is case when a name is displaying with "xx")</li>
<li>Like you want, I removed the files test.* and I add my suggestion to the grimaldi.gwf
</ul>

alifirat committed:
Fix the condition in order to cancel links only for person with names "x x"

alifirat committed:
Fix the condition in order to cancel links only for person with names "x x"

alifirat commented:
I update the condition in the person field "is_obfuscated". With the previous version, all links was canceled because I forgot to add the condition about the age of a person. Now, only names with "x x" are canceled, sorry for forgetting.

fablhx commented:
@alifirat, @swdes, I'm not sure the is_obfuscated is a good idea. I think just testing (like in the first version) "xx" = "x x" is better, because if you add a 'variable' is_obfuscated, you'll need to add it in every
template environment that needs it.

fablhx commented:
@alifirat, if you still want to keep is_obfuscated, you should change your condition and use p_auth, it's a more precise version of your fast_auth_age.

fablhx commented:
@alifirat, I think your patch is not exactly what I thought about. In particular, I think it's not woking with pages that are not generated through templates.

alifirat commented:
Hello @ipfix  and thank's for your answers. Like you see, I had two version of testing "x x". I changed the first version after some suggestions and I'm wondering if it can be the time to use "is_obfuscated" instead of "xx = x x" (simple suggestion). In fact, I also understand your remark because I saw a lot of times the expression "xx = x x".

The idea of the patch that I submitted is that for remove private links, It's just  to change the displaying of the HTML rendering. When I searched how to change that, I can see that the displaying is managed by the templates. Also, I had a use case in which I can test the code that I changed and like in your remarks, It's possible to have pages not generated through templates. So my question is : like in the issue that your opened, do you have a use case in which I can see the problem that you raised in your last comment ?

alifirat committed:
Change fash_auth_age by p_auth

alifirat committed:
Merge branch 'cancel_private_links##issue333' of github.com:alifirat/geneweb into cancel_private_links##issue333

Conflicts:
	src/perso.ml

alifirat committed:
Add condition in function long_display_person to cancel private links

swdes commented:
Hello @alifirat 
Your implementation seems to work well but only on the first template ("default")
It should work on every template (from templa to templh).

NB: to switch to another template, use the footer menu bar.
![image](https://cloud.githubusercontent.com/assets/5917598/8822879/d3d14568-3068-11e5-998b-f631e7eec180.png)

alifirat committed:
Cancel private links "x x" for the templates templa to templh + show "unknown_sex.png" for person "x x"

alifirat commented:
@swdes It's done for the templates templa to templh (and only them like in your comment).

Also, I added the following thing : when a user wants to cancel private links, I didn't show the images "male.png" or "female.png" but "sexunknown.png" . I found that prettier than a name "x x" with a male or female sex image.

swdes commented:
Thank you @alifirat

dominique95880 commented:
@ipfix said : I think your patch is not exactly what I thought about. In particular, I think it's not woking with pages that are not generated through templates.
@alifrat said : So my question is : like in the issue that your opened, do you have a use case in which I can see the problem that you raised in your last comment ?

and I answer for him
- List of Surname, example on  http://gw.geneanet.org/testgeneanet1?lang=en;m=N;v=geneanet&tri=A
- List of Firstname, example on http://gw.geneanet.org/testgeneanet1?k=&lang=en&m=P&o=A&tri=A
- Advanced request, example on http://gw.geneanet.org/testgeneanet1?lang=en&m=AS_OK;surname=GENEANET
- Family Tree (descendants), exemple on http://gw.geneanet.org/testgeneanet1?lang=en&m=D&p=arriere+grand+pere&n=geneanet&sosab=10&color=&t=T&v=3
- and so on

Dominique

alifirat commented:
Hello @dominique95880 

Thank your for the list of use case. I would like to ask a question about it : In every use case in your list, can I set the configuration to the private mode in order to see names "x x" and so edit some code to cancel links like in the use case of @ipfix ?

Everybody would appreciate to have a fix working on the all software and not only for pages generated by templates. For this reason,  do you have some advices to give ? 

Ali-Firat.

fablhx commented:
This PR is not complete for #333 #342. I close it.
